crates/autopilot/src/run_loop.rs

```diff
@@ -212,20 +261,103 @@ impl RunLoop {
+            let solved_order_uids: HashSet<_> = solution.orders().keys().cloned().collect();
+            self.in_flight_orders
+                .lock()
                 .await
-            {
-                Ok(()) => Metrics::settle_ok(driver, submission_start.elapsed()),
-                Err(err) => {
-                    Metrics::settle_err(driver, submission_start.elapsed(), &err);
-                    tracing::warn!(?err, driver = %driver.name, "settlement failed");
+                .extend(solved_order_uids.clone());
+
+            let solution_id = solution.id();
+            let driver_ = driver.clone();
+            let self_ = self.clone();
+
+            let settle_fut = async move {
+                tracing::info!(driver = %driver_.name, "settling");
+                let submission_start = Instant::now();
+
+                match self_
+                    .settle(
+                        driver_.clone(),
+                        solution_id,
+                        solved_order_uids,
+                        auction_id,
+                        block_deadline,
+                    )
+                    .await
+                {
+                    Ok(()) => {
+                        Metrics::settle_ok(&driver_, submission_start.elapsed());
+                    }
+                    Err(err) => {
+                        Metrics::settle_err(&driver_, submission_start.elapsed(), &err);
+                        tracing::warn!(?err, driver = %driver_.name, "settlement failed");
+                    }
                 }
+                Metrics::single_run_completed(single_run_start.elapsed());
+            };
+
+            let sender = self.get_settlement_queue_sender(&driver.name).await;
+            if let Err(err) = sender.send(Box::pin(settle_fut)) {
+                tracing::warn!(driver = %driver.name, ?err, "failed to send settle future to queue");
             }
+        }
+    }
 
-            Metrics::single_run_completed(single_run_start.elapsed());
+    /// Retrieves or creates the settlement queue sender for a given driver.
+    ///
+    /// This function ensures that there is a settlement execution queue
+    /// associated with the specified `driver_name`. If a queue already
+    /// exists, it returns the existing sender. If not, it creates a new
+    /// queue, starts a background task to process settlement futures
+    /// sequentially, and returns the new sender.
+    ///
+    /// The background task processes futures from the queue one by one,
+    /// guaranteeing FIFO execution order for settlements per driver. It
+    /// also implements an idle timeout: if no new futures are received within
+    /// `IDLE_TIMEOUT`, the background task terminates, and the queue is
+    /// removed.
+    async fn get_settlement_queue_sender(
+        self: &Arc<Self>,
+        driver_name: &str,
+    ) -> UnboundedSender<BoxFuture<'static, ()>> {
+        const IDLE_TIMEOUT: Duration = Duration::from_secs(300);
```

Comment:
Having this timeout mechanism brings a lot of complexity with it.
I think it's okay to to see 1 queue per solver as a fixed cost and just keep it around forever.
This should be fine as the number of queues is very low and statically known as opposed to theoretically unbounded for things like price caches for example.